plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -257,4 +264,25 @@ protected static List<Column> toHiveColumns(List<NestedField> columns)
                         Optional.empty()))
                 .collect(toImmutableList());
     }
+
+    protected CommitStatus checkCommitStatus(String metadataLocation)
+    {
+        AtomicReference<CommitStatus> status = new AtomicReference<>(CommitStatus.UNKNOWN);
+
+        Tasks.foreach(metadataLocation)
```

Comment:
That's iceberg way of doing retries. we don't practise this in trino
we typically use `net.jodah.failsafe.Failsafe`

Reply:
We are using `Tasks.forEach` in `AbstractIcebergTableOperations#refreshFromMetadataLocation`

https://github.com/krvikash/trino/blob/68d3da6770f3195dec19a6812797210d8b4c4618/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java#L224-L229

Reply:
i know. that doesn't mean it's the right thing to do